File: litedb-java/src/main/java/com/litedb/relational/Column.java
package com.litedb.relational;

/**
 * Column — a typed column in a table schema.
 *
 * Phase 1 stores all column values as strings; typed, order-preserving encoding
 * (so numeric range queries sort correctly) is a later phase.
 */
public final class Column {
    public final String name;
    public final String type;   // INT | TEXT | FLOAT | BOOL

    public Column(String name, String type) {
        this.name = name;
        this.type = type;
    }

    @Override
    public String toString() {
        return name + " " + type;
    }
}

File: litedb-python/column.py
"""column.py — a typed column in a table schema."""


class Column:
    __slots__ = ("name", "type")

    def __init__(self, name: str, type_: str):
        self.name = name
        self.type = type_   # INT | TEXT | FLOAT | BOOLEAN

    def __repr__(self):
        return f"{self.name} {self.type}"
